__tests__/components/AnonymousChat/child/ChatItemMyTextV2.test.tsx

```diff
@@ -1,43 +1,178 @@
+  };
+});
+
+afterEach(() => {
+  jest.restoreAllMocks();
+});
+
+describe('TESTING ChatItemMyTextV2', () => {
   const avatar = (
     <FastImage
       style={{height: 24, width: 24, borderRadius: 12}}
       source={{uri: DEFAULT_PROFILE_PIC_PATH}}
     />
   );
 
-  it('RENDER not continuous should match snapshot', () => {
-    const {toJSON} = render(
-      <ChatItemMyTextV2
-        avatar={avatar}
-        username="username"
-        isContinuous={false}
-        message="message"
-        time="time"
-      />
+  it('should call setReplyPreview when swiped', () => {
+    const setReplyPreview = jest.fn();
+    const contextValue = {
+      chat: [{replyTarget: null}, setReplyPreview],
+      profile: [{myProfile: {}}]
+    };
+
+    const {getByTestId} = render(
+      <Context.Provider value={contextValue}>
+        <ChatItemMyTextV2
+          avatar={avatar}
+          username="username"
+          isContinuous={false}
+          message="message"
+          time="time"
+          chatType="SIGNED"
+          messageType="regular"
+          data={{}}
+        />
+      </Context.Provider>
     );
 
-    expect(toJSON()).toMatchSnapshot();
+    act(() => {
+      fireEvent(getByTestId('swipeable'), 'onSwipeableOpen', 'left');
+    });
+
+    expect(setReplyPreview).toHaveBeenCalled();
   });
 
-  it('RENDER continuous should match snapshot', () => {
-    const {toJSON} = render(
-      <ChatItemMyTextV2
-        avatar={avatar}
-        username="username"
-        isContinuous={true}
-        message="message"
-        time="time"
-      />
+  it('should display the username and time when isContinuous is false', () => {
+    const contextValue = {
+      chat: [{replyTarget: null}, jest.fn()],
+      profile: [{myProfile: {}}]
+    };
+
+    const {getByText} = render(
+      <Context.Provider value={contextValue}>
+        <ChatItemMyTextV2
+          avatar={avatar}
+          username="username"
+          isContinuous={false}
+          message="message"
+          time="time"
+          chatType="SIGNED"
+          messageType="regular"
+          data={{}}
+        />
+      </Context.Provider>
     );
 
-    expect(toJSON()).toMatchSnapshot();
+    expect(getByText('username')).toBeTruthy();
+    expect(getByText('time')).toBeTruthy();
+  });
+
+  it('should not display the username and time when isContinuous is true', () => {
+    const contextValue = {
+      chat: [{replyTarget: null}, jest.fn()],
+      profile: [{myProfile: {}}]
+    };
+
+    const {queryByText} = render(
+      <Context.Provider value={contextValue}>
+        <ChatItemMyTextV2
+          avatar={avatar}
+          username="username"
+          isContinuous={true}
+          message="message"
+          time="time"
+          chatType="SIGNED"
+          messageType="regular"
+          data={{}}
+        />
+      </Context.Provider>
+    );
+
+    expect(queryByText('username')).toBeNull();
+    expect(queryByText('time')).toBeNull();
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This test case checks if the username and time are not displayed when `isContinuous` is true. Similar to the previous comment, it does not check if the avatar is displayed correctly. It would be more comprehensive to also check the avatar.



<!-- This is an auto-generated comment by OSS CodeRabbit -->